airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -194,12 +194,26 @@ def _read(
         # to prevent it from showing in the UI.
         def concat_logs(lines):
             log_range = (len(lines) - 1) if lines[-1].message == self.end_of_log_mark.strip() else len(lines)
-            return '\n'.join([lines[i].message for i in range(log_range)])
+            return '\n'.join([self._format_msg(lines[i]) for i in range(log_range)])
 
         message = [(host, concat_logs(hosted_log)) for host, hosted_log in logs_by_host]
 
         return message, metadata
 
+    def _format_msg(self, log_line):
+        """Format ES Record to match settings.LOG_FORMAT when used with json_format"""
+        # Using formatter._style.format makes it future proof i.e.
+        # if we change the formatter style from '%' to '{' or '$', this will still work
+        if self.json_format:
+            try:
+                # pylint: disable=protected-access
+                return self.formatter._style.format(_ESJsonLogFmt(**log_line.to_dict()))
+            except Exception:  # noqa pylint: disable=broad-except
```

Comment:
What is the reason for exception?

Reply:
For now, I can only foresee `KeyError` here i.e. if log_format has a key that is not present in ES logs:
https://github.com/apache/airflow/blob/69d6d0239f470ac75e23160bac63408350c1835a/airflow/config_templates/default_airflow.cfg#L289

Reply:
But kept a broad-except to avoid any unforeseen exception but it is just a safeguard and keeps the same behavior as now i.e. return `log_line.message`